### ISSUE TITLE
NotImplementedException in change feed handler

### ISSUE THREAD

**Issue by mortenbock** [Azure/azure-cosmos-dotnet-v3#4093]
Using the 3.35.3 nuget package, I'm getting this in my logs;

```
System.NotImplementedException: The method or operation is not implemented. at 
Microsoft.Azure.Cosmos.ChangeFeed.ChangeFeedEstimatorIterator.ChangeFeedEstimatorEmptyFeedResponse.get_Headers() at 
Microsoft.Azure.Cosmos.OpenTelemetryResponse`1..ctor(FeedResponse`1 responseMessage) at 
Microsoft.Azure.Cosmos.ChangeFeed.ChangeFeedEstimatorIterator.<>c.<ReadNextAsync>b__20_1(FeedResponse`1 response) at 
Microsoft.Azure.Cosmos.ClientContextCore.RunWithDiagnosticsHelperAsync[TResult](String containerName, String databaseName, OperationType operationType, ITrace trace, Func`2 task, Func`2 openTelemetry, String operationName, RequestOptions requestOptions)
```

Thrown from here;
https://github.com/Azure/azure-cosmos-dotnet-v3/blob/258d960ae3caa3ad989f60d5e656544c35006d0a/Microsoft.Azure.Cosmos/src/ChangeFeedProcessor/ChangeFeedEstimatorIterator.cs#L405

**Comment by ealsur** [comment]
@mortenbock The stack trace points to OpenTelemetry, are you using 3.35.3 or 3.35.3-preview?

**Comment by ealsur** [comment]
@sourabh1007 Is OpenTelemetry in this case always accessing the Headers? This will never work on the Estimator, the responses are not Network responses.

**Comment by mortenbock** [comment]
@ealsur We were using it, but I'm migrating away from the -preview packages now to disable it, because it was incredibly chatty, but that is a discussion for a separate issue :)

**Comment by ealsur** [comment]
@mortenbock Please provide feedback about OpenTelemetry integration in a separate Issue, it's very valuable to us.

**Comment by ealsur** [comment]
@mortenbock Looking at this more in detail:

https://github.com/Azure/azure-cosmos-dotnet-v3/blob/455656753e115d1667bbaa0ae8689a2f3686a6d2/Microsoft.Azure.Cosmos/src/ChangeFeedProcessor/ChangeFeedEstimatorIterator.cs#L169-L174

This only happens if there is no Change Feed Processor running, is this expected in your case?

**Comment by mortenbock** [comment]
@ealsur I _think_ in our case, the code for the Change Feed Processor was active, but the container that it was listening for changes from was empty/new, so there were no documents in there.

I don't know if that covers the same case?

**Comment by ealsur** [comment]
@mortenbock No, the condition for this to happen is the Lease Container is empty. This means, a Processor was never running/never started on that Lease Container. Leases are created when the Processor starts for the first time. The above `if` happens if there are no lease documents at all.

Which means either there is a misconfiguration (your estimator is not pointing to the lease container used by your Processor) or there is a fundamental failure (you are using the Estimator without a Processor to monitor).

This does not remove the need to solve the exception in Open Telemetry, but it is pointing to something not being correct in the way you are using the Estimator.

**Comment by mortenbock** [comment]
@ealsur Thanks you for the feedback. Looking at our code, we are starting the estimator before the processor like this:

    await _changeFeedEstimator.StartAsync();
    await _changeFeedProcessor.StartAsync();

So I guess I could swap the order of those two, so the estimator does not start before the processor has created a lease document the first time the application deploys?

If we were to deploy the estimator separately, would there be a good way to check if a lease exists before starting the estimator?

**Comment by bartelink** [comment]
There's no correlation between when you start the processor or estimator and when the lease/checkpoint document/item is written - i.e., one doesn't wait for the other, and there is no guaranteed max time that one could sleep for.

Question: If the estimator is left running, will it work once a lease has been established, or does the absence of one prevent it from starting its normal cycle?

... And a bump for the programmatic way to trigger _deletion_ or rewinding of the checkpoint request: https://github.com/Azure/azure-cosmos-dotnet-v3/issues/510
